### PR TITLE
LG-3983: Updates Badge border-radius

### PR DESCRIPTION
**Pull request by bruugey**
## ✍️ Proposed changes

Updates Badge border-radius

🎟 _Jira ticket:_ [LG-3983](https://jira.mongodb.org/browse/LG-3983)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

**Review by sandynguyenn — APPROVED** [review]
LGTM

File: .changeset/many-cheetahs-hear.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/badge': patch
```
Comment (shaneeza) [inline .changeset/many-cheetahs-hear.md:2]: Since the badge will look different, should it be a minor change?

Comment (bruugey) [reply .changeset/many-cheetahs-hear.md:2]: I don't think so - but what do you think?

Comment (bruugey) [reply .changeset/many-cheetahs-hear.md]: Nvm. UPdated.

Comment (shaneeza) [reply .changeset/many-cheetahs-hear.md]: What made you switch? Honestly, I'm not so sure myself. The documents are not that helpful.

Comment (bruugey) [reply .changeset/many-cheetahs-hear.md]: Talked it out with Rob and Dave. It's not a bug, so it's not really a patch.

**Review by shaneeza — APPROVED** [review]

**Pull request merged**